src/background/actions/assessment-card-selection-action-creator.ts

```diff
@@ -0,0 +1,93 @@
+import { getStoreStateMessage, Messages } from '../../common/messages';
+import { Interpreter } from '../interpreter';
+import { TelemetryEventHandler } from '../telemetry/telemetry-event-handler';
+import {
+    AssessmentCardSelectionPayload,
+    AssessmentCardToggleVisualHelperPayload,
+    AssessmentExpandCollapsePayload,
+    AssessmentSingleRuleExpandCollapsePayload,
+} from './action-payloads';
+
+export class AssessmentCardSelectionActionCreator {
+    constructor(
+        private readonly interpreter: Interpreter,
+        private readonly assessmentCardSelectionActions: AssessmentCardSelectionActions,
+        private readonly telemetryEventHandler: TelemetryEventHandler,
+    ) {}
+
+    public registerCallbacks(): void {
+        this.interpreter.registerTypeToPayloadCallback(
+            Messages.AssessmentCardSelection.CardSelectionToggled,
+            this.onAssessmentCardSelectionToggle,
+        );
+        this.interpreter.registerTypeToPayloadCallback(
+            Messages.AssessmentCardSelection.RuleExpansionToggled,
+            this.onRuleExpansionToggle,
+        );
+        this.interpreter.registerTypeToPayloadCallback(
+            getStoreStateMessage(StoreNames.AssessmentCardSelectionStore),
+            this.onGetCurrentState,
+        );
+        this.interpreter.registerTypeToPayloadCallback(
+            Messages.AssessmentCardSelection.ToggleVisualHelper,
+            this.onToggleVisualHelper,
+        );
+        this.interpreter.registerTypeToPayloadCallback(
+            Messages.AssessmentCardSelection.ExpandAllRules,
+            this.onExpandAllRules,
+        );
+        this.interpreter.registerTypeToPayloadCallback(
+            Messages.AssessmentCardSelection.CollapseAllRules,
+            this.onCollapseAllRules,
+        );
+    }
+
+    private onGetCurrentState = async (): Promise<void> => {
+        await this.assessmentCardSelectionActions.getCurrentState.invoke(null);
+    };
+
+    private onAssessmentCardSelectionToggle = async (
+        payload: AssessmentCardSelectionPayload,
+    ): Promise<void> => {
+        await this.assessmentCardSelectionActions.toggleCardSelection.invoke(payload);
+        this.telemetryEventHandler.publishTelemetry(
+            TelemetryEvents.CARD_SELECTION_TOGGLED,
+            payload,
+        );
+    };
+
+    private onRuleExpansionToggle = async (
+        payload: AssessmentSingleRuleExpandCollapsePayload,
+    ): Promise<void> => {
+        await this.assessmentCardSelectionActions.toggleRuleExpandCollapse.invoke(payload);
+        this.telemetryEventHandler.publishTelemetry(
+            TelemetryEvents.RULE_EXPANSION_TOGGLED,
+            payload,
+        );
+    };
+
+    private onToggleVisualHelper = async (
+        payload: AssessmentCardToggleVisualHelperPayload,
+    ): Promise<void> => {
+        await this.assessmentCardSelectionActions.toggleVisualHelper.invoke(null);
```

Comment:
Need to pipe payload through (also on line 85 and 90). Seems like tests should have caught this- might mean we have to update the tests as well?